### DISCUSSION TITLE
Feature Request: Function to convert TIMESTAMP -> MILISECONDS_SINCE_EPOCH

### DISCUSSION THREAD

**Discussion by 2-5** [duckdb/duckdb#7563] [Ideas]
DuckDB has no function to convert a TIMESTAMP to miliseconds since epoch.

Existing functions:

```python
epoch:        TIMESTAMP -> SECONDS_SINCE_EPOCH
to_timestamp: SECONDS_SINCE_EPOCH -> TIMESTAMP

# NOT A TYPO, inconsistent naming
epoch_ms:     MILISECONDS_SINCE_EPOCH -> TIMESTAMP
?????:        TIMESTAMP -> MILISECONDS_SINCE_EPOCH
```

**Comment by hawkfish** [comment]
Wow that's unfortunate. Plus we should be dealing with µs as that is our internal time base.

**Comment by hawkfish** [comment]
I think `make_timestamp` can be overloaded so that a single argument is interpreted as microseconds.

**Comment by esemeniuc** [comment]
I would love to see this as well!

**Comment by mustafahasankhan** [comment]
epoch_ms can be used to do both!
It converts timestamp to milliseconds since epoch and vice-versa!

**Reply by esemeniuc** [reply]
I see it here, thank you :) 

https://duckdb.org/docs/sql/functions/timestamp#:~:text=00%3A00%3A00-,epoch_ms(timestamp),-Return%20the%20total

**Reply by esemeniuc** [reply]
For querying influx csv data, you can do something like
```
SELECT time_bucket(INTERVAL '3h', epoch_ms(TIME)) AS bucket, round(AVG(value))
FROM 'influx.csv'
GROUP BY bucket
ORDER BY bucket ASC;
```

**Comment by hawkfish** [comment]
Also `date_part('epoch', ts)` now returns a `DOUBLE`, which is to be preferred as `epoch_ms` is being deprecated.
